### ISSUE TITLE
RtContainersTestCase.java:33-35: Write unit tests for all...

### ISSUE THREAD

**Issue by 0pdd** [amihaiemil/docker-java-api#47]
The puzzle `26-a8980bc9` from #26 has to be resolved:

https://github.com/amihaiemil/docker-java-api/blob/f5a14a642034a8bcde09e708cb5c27564061dfab/src/test/java/com/amihaiemil/docker/RtContainersTestCase.java#L33-L35

The puzzle was created by amihaiemil on 13-Mar-18. 

Estimate: 30 minutes, role: DEV.

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0crat** [comment]
Job [#47](https://github.com/amihaiemil/docker-java-api/issues/47) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
The job [#47](https://github.com/amihaiemil/docker-java-api/issues/47) assigned to @llorllale[/z](https://www.0crat.com/u/llorllale), here is [why](https://www.0crat.com/footprint/G6LPQQV2P/731613202). The budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by llorllale** [comment]
@amihaiemil what's the url?

```java
@param container Json config as specified in the API's docs.
```

Would be nice if we added to the javadocs:

```java
@see <a href="api_doc_url">Docker API</a>
```

**Comment by amihaiemil** [comment]
@llorllale right; this is the URL: https://docs.docker.com/engine/api/v1.30/#operation/ContainerCreate

The API docs is for 1.30; I guess it won't change that much in future versions. Besides, whoever knows how to read it will also know to check the right version they need :)

**Comment by llorllale** [comment]
@amihaiemil I'm think about creating `Response` that implements `HttpResponse` in order to support tests. WDYT?

**Comment by llorllale** [comment]
@amihaiemil here's an example of what this might all look like:

```java
    /**
     * Should create container if request valid and no error is returned from
     * the Docker API.
     * 
     * @throws Exception unexpected
     */
    @Test
    public void createContainerFromImageName() throws Exception {
        MatcherAssert.assertThat(
            new RtContainers(
                new AssertRequest(
                    new Response(
                        HttpStatus.SC_CREATED, "{ \"Id\": \"df2419f4\" }"
                    ),
                    new Condition(
                        "The 'Content-Type' header must be set.",
                        req -> req.getHeaders("Content-Type").length > 0
                    ),
                    new Condition(
                        "Content-Type must be 'application/json'.",
                        // @checkstyle LineLength (1 line)
                        req -> "application/json".equals(req.getHeaders("Content-Type")[0].getValue())
                    ),
                    new Condition(
                        "Resource path must be /containers/create",
                        // @checkstyle LineLength (1 line)
                        req -> req.getRequestLine().getUri().endsWith("/containers/create")
                    )
                ), URI.create("http://localhost/test")
            ).create("some_image"),
            Matchers.notNullValue()
        );
    }
```

**Comment by amihaiemil** [comment]
@llorllale Yeah, seems like a good idea :)

**Comment by 0pdd** [comment]
The puzzle `26-a8980bc9` has disappeared from the source code, that's why I closed this issue.

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#53](https://github.com/amihaiemil/docker-java-api/issues/53), [#54](https://github.com/amihaiemil/docker-java-api/issues/54) are still not solved.

**Comment by 0crat** [comment]
Order was finished: +30 points just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
The job [#47](https://github.com/amihaiemil/docker-java-api/issues/47) is now out of scope

**Comment by 0pdd** [comment]
@0pdd the puzzle [#54](https://github.com/amihaiemil/docker-java-api/issues/54) is still not solved; solved: [#53](https://github.com/amihaiemil/docker-java-api/issues/53).

**Comment by 0pdd** [comment]
@0pdd all 2 puzzles are solved here: [#53](https://github.com/amihaiemil/docker-java-api/issues/53), [#54](https://github.com/amihaiemil/docker-java-api/issues/54).
